### PR TITLE
_validate() moved to helpers folder

### PR DESCRIPTION
**Pull request by AtharvSan**

### CODE REVIEW

**Comment by tabshaikh** [comment]
we might not want to do it like this but do a contract call instead

File: packages/contracts/src/libraries/helpers/Validation.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.25;
+
+import "../../interfaces/ICrossL2Inbox.sol";
+import {Predeploys} from "../Predeploys.sol";
+import {ICrossL2Prover} from "../../interfaces/ICrossL2Prover.sol";
+
+contract Validation {
+    ICrossL2Prover private crossL2Prover;
```
Comment (tabshaikh) [inline packages/contracts/src/libraries/helpers/Validation.sol:9]: also where is the prover being set

**Comment by tabshaikh** [comment]
please modify the deploy.s.sol to also then deploy the validation contract

File: packages/contracts/src/libraries/helpers/Validation.sol (hunk 1 above)
Comment (0xumarkhatab) [reply packages/contracts/src/libraries/helpers/Validation.sol:9]: Having a look at it

Comment (0xumarkhatab) [reply packages/contracts/src/libraries/helpers/Validation.sol:9]: yea @AtharvSan add a constructor to set crossL2Prover

File: packages/contracts/src/libraries/helpers/Validation.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    enum ValidationMode {
+        CUSTOM,
+        CROSS_L2_INBOX,
+        CROSS_L2_PROVER_EVENT,
+        CROSS_L2_PROVER_RECEIPT
+    }
+
+    function _validate(
```
Comment (0xumarkhatab) [inline packages/contracts/src/libraries/helpers/Validation.sol:18]: make validate a callable external function .. or you know what can also be made a library , but then you have to pass the crossL2Prover along side `validate` function call..
let it be a contract , and make external calls each time to validate stuff

**Pull request closed without merging**